### PR TITLE
pkg/cluster: Use Action for fixupClusterSPObjectID

### PR DESCRIPTION
**Pull request by mbarnes**
### Which issue this PR addresses:

[[ARO-1919] AAD: Migrate ARO-RP Azure AD Graph -> Microsoft Graph](https://issues.redhat.com/browse/ARO-1919)

### What this PR does / why we need it:

Minor follow-up to https://github.com/Azure/ARO-RP/pull/1970 where @bennerv requested this in his review.

The `fixupClusterSPObjectID` step in PUCM uses only the [MS Graph client](https://pkg.go.dev/github.com/microsoftgraph/msgraph-sdk-go#GraphServiceClient) (no ARM calls), and the client already handles HTTP 401 Unauthorized errors by refreshing its authorization token. So there's no need to use the [AuthorizationRetryingAction](https://github.com/Azure/ARO-RP/blob/1ee42feb78ae64ca694bf832be3ee17bde8cbea7/pkg/util/steps/refreshing.go#L21-L24) wrapper.
    
The 401 Unauthorized error handling logic for the MS Graph client is in the [github.com/microsoft/kiota-http-go](https://github.com/microsoft/kiota-http-go/blob/main/nethttp_request_adapter.go#L173-L198) module.
   

### Test plan for issue:

- Create an ARO cluster.
- Run `az aro update --refresh-credentials` on it.
- Verify it completes successfully.

### Is there any documentation that needs to be updated for this PR?

No, this is just an internal change.

### CODE REVIEW

**Comment by mbarnes** [comment]
Holding until we can sort out some permission issues in the dev subscription.  It's blocking a successful test.

```
$ az aro update --refresh-credentials --name $CLUSTER --resource-group $RESOURCEGROUP
...
Traceback (most recent call last):
  File "/home/mbarnes/git/github.com/mbarnes/ARO-RP/python/az/aro/azext_aro/custom.py", line 552, in cluster_application_update
    client_secret = aad.add_password(app)
  File "/home/mbarnes/git/github.com/mbarnes/ARO-RP/python/az/aro/azext_aro/_aad.py", line 55, in add_password
    cred = self.client.application_add_password(obj_id, {})
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/role/_msgrpah/_graph_client.py", line 132, in application_add_password
    result = self._send("POST", "/applications/{id}/addPassword".format(id=id), body=body)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/command_modules/role/_msgrpah/_graph_client.py", line 55, in _send
    raise GraphError(ex.response.json()['error']['message'], ex.response) from ex
azure.cli.command_modules.role._msgrpah._graph_client.GraphError: Insufficient privileges to complete the operation.
```

**Comment by bennerv** [comment]
@mbarnes were you ever able to get around the graph issues after the Azure AD admin made changes?

**Comment by dkeohane** [comment]
@mbarnes anything needed for this to get merged? Can we look to remove the ready-for-review label if its gonna stay on hold?

**Comment by mbarnes** [comment]
I guess it could use re-testing since the PR is a year-and-a-half old now.  It's just a minor cleanup to remove some unnecessary logic around MS Graph calls.  Or we could wait until ARO is fully migrated to [azure-sdk-for-go](https://github.com/Azure/azure-sdk-for-go), at which time [AuthorizationRetryingAction](https://github.com/Azure/ARO-RP/blob/1ee42feb78ae64ca694bf832be3ee17bde8cbea7/pkg/util/steps/refreshing.go#L21-L24) should die naturally.

File: pkg/cluster/install.go [hunk 1]
```diff
@@ -46,11 +46,7 @@ func (m *manager) adminUpdate() []steps.Step {
 		steps.Action(m.initializeKubernetesClients), // must be first
 		steps.Action(m.ensureBillingRecord),         // belt and braces
 		steps.Action(m.ensureDefaults),
-
-		// TODO: this relies on an authorizer that isn't exposed in the manager
-		// struct, so we'll rebuild the fpAuthorizer and use the error catching
-		// to advance
-		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.fixupClusterSPObjectID),
```
Comment (mociarain) [inline pkg/cluster/install.go:53]: This comment has changed: https://github.com/azure/ARO-RP/blob/176a68d4b8494cbbdf696334a36823c1c468d490/pkg/cluster/install.go#L197

Is this change still good?

**Review by mociarain — COMMENTED** [review]
Small question.

Note if the change is still good I think a new PR would be easier than rebasing.

**Comment by mociarain** [comment]
Closed in favour of: https://github.com/Azure/ARO-RP/pull/3694

**Pull request closed without merging**